modules/reactor/test/tests/meshgenerators/patterned_cartesian_mesh_generator/tests

```diff
@@ -0,0 +1,205 @@
+    []
+    [err_out_of_range_pattern]
+      type = 'RunException'
+      input = 'patterned_2d.i'
+      cli_args = 'Mesh/pattern/pattern_boundary=none
+                  Mesh/pattern/pattern="0 0 0;0 1 0;0 2 0"
+                  --mesh-only "patterned_err.e"'
+      expect_err = 'Elements of this parameter must be smaller than the length of input_name.'
+      requirement ='The system shall throw an error if the pattern includes input cartesian mesh indices that exceed number of elements of inputs.'
+    []
+    [err_block_id]
+      type = 'RunException'
+      input = 'patterned_2d.i'
+      cli_args = 'Mesh/pattern/background_intervals=2 Mesh/pattern/duct_block_ids=25
+                  Mesh/pattern/background_block_name="assem_block"
+                  Mesh/pattern/sqaure_size=22 Mesh/pattern/duct_sizes=21 Mesh/pattern/duct_intervals=2
+                  --mesh-only "patterned_err.e"'
+      expect_err = 'This parameter and duct_block_ids must be provided simultaneously.'
+      requirement ='The system shall throw an error if background_block_id and duct_block_ids are not provided simultaneously if the cartesian mesh has ducts.'
+    []
+    [err_block_name]
+      type = 'RunException'
+      input = 'patterned_2d.i'
+      cli_args = 'Mesh/pattern/background_intervals=2 Mesh/pattern/duct_block_names="duct1"
+                  Mesh/pattern/sqaure_size=22 Mesh/pattern/duct_sizes=21 Mesh/pattern/duct_intervals=2
+                  --mesh-only "patterned_err.e"'
+      expect_err = 'This parameter and duct_block_names must be provided simultaneously.'
+      requirement ='The system shall throw an error if background_block_name and duct_block_names are not provided simultaneously if the cartesian mesh has ducts.'
+    []
+    [err_duct_ids]
+      type = 'RunException'
+      input = 'patterned_2d.i'
+      cli_args = 'Mesh/pattern/background_intervals=2 Mesh/pattern/duct_block_ids=20 Mesh/pattern/background_block_id=15
+                  Mesh/pattern/sqaure_size=22 Mesh/pattern/duct_sizes="21.0 21.5" Mesh/pattern/duct_intervals="2 2"
+                  --mesh-only "patterned_err.e"'
+      expect_err = 'This parameter, if provided, must have a length equal to length of duct_sizes.'
+      requirement ='The system shall throw an error if length of duct_block_ids is not consistent with number of cartesian ducts.'
+    []
+    [err_duct_names]
+      type = 'RunException'
+      input = 'patterned_2d.i'
+      cli_args = 'Mesh/pattern/background_intervals=2 Mesh/pattern/duct_block_names="duct1" Mesh/pattern/background_block_name="background"
+                  Mesh/pattern/sqaure_size=22 Mesh/pattern/duct_sizes="21.0 21.5" Mesh/pattern/duct_intervals="2 2"
+                  --mesh-only "patterned_err.e"'
+      expect_err = 'This parameter, if provided, must have a length equal to length of duct_sizes.'
+      requirement ='The system shall throw an error if length of duct_block_names is not consistent with number of cartesian ducts.'
```

Comment:
```suggestion
      requirement ='The system shall throw an error if number of duct names specified is not consistent with number of cartesian ducts.'
```
try to avoid naming parameters in requirements please